src/server/json_family_test.cc

```diff
@@ -155,8 +152,280 @@ TEST_F(JsonFamilyTest, StrLen) {
   EXPECT_THAT(resp, IntArg(2));
+
+  resp = Run({"JSON.STRLEN", "json", "$.d.*"});
+  ASSERT_THAT(resp, ArgType(RespExpr::ARRAY));
+  EXPECT_THAT(resp.GetVec(), ElementsAre(ArgType(RespExpr::NIL), IntArg(1), ArgType(RespExpr::NIL)));
+}
+
+TEST_F(JsonFamilyTest, ObjLen) {
+  string json = R"(
+    {"a":{}, "b":{"a":"a"}, "c":{"a":"a", "b":"bb"}, "d":{"a":1, "b":"b", "c":{"a":3,"b":4}}, "e":1}
+  )";
+
+  auto resp = Run({"set", "json", json});
+  ASSERT_THAT(resp, "OK");
+
+  resp = Run({"JSON.OBJLEN", "json", "$.a"});
+  EXPECT_THAT(resp, IntArg(0));
+
+  resp = Run({"JSON.OBJLEN", "json", "$.a.*"});
+  EXPECT_THAT(resp, ArgType(RespExpr::NIL_ARRAY));
+
+  resp = Run({"JSON.OBJLEN", "json", "$.b"});
+  EXPECT_THAT(resp, IntArg(1));
+
+  resp = Run({"JSON.OBJLEN", "json", "$.b.*"});
+  EXPECT_THAT(resp, ArgType(RespExpr::NIL));
+
+  resp = Run({"JSON.OBJLEN", "json", "$.c"});
+  EXPECT_THAT(resp, IntArg(2));
+
+  resp = Run({"JSON.OBJLEN", "json", "$.c.*"});
+  ASSERT_THAT(resp, ArgType(RespExpr::ARRAY));
+  EXPECT_THAT(resp.GetVec(), ElementsAre(ArgType(RespExpr::NIL), ArgType(RespExpr::NIL)));
+
+  resp = Run({"JSON.OBJLEN", "json", "$.d"});
+  EXPECT_THAT(resp, IntArg(3));
+
+  resp = Run({"JSON.OBJLEN", "json", "$.d.*"});
+  ASSERT_THAT(resp, ArgType(RespExpr::ARRAY));
+  EXPECT_THAT(resp.GetVec(), ElementsAre(ArgType(RespExpr::NIL), ArgType(RespExpr::NIL), IntArg(2)));
+
+  resp = Run({"JSON.OBJLEN", "json", "$.*"});
+  ASSERT_THAT(resp, ArgType(RespExpr::ARRAY));
+  EXPECT_THAT(resp.GetVec(), ElementsAre(IntArg(0), IntArg(1), IntArg(2), IntArg(3), ArgType(RespExpr::NIL)));
+}
+
+TEST_F(JsonFamilyTest, ArrLen) {
+  string json = R"(
+    [[], ["a"], ["a", "b"], ["a", "b", "c"]]
+  )";
+
+  auto resp = Run({"set", "json", json});
+  ASSERT_THAT(resp, "OK");
+
+  resp = Run({"JSON.ARRLEN", "json", "$[*]"});
+  ASSERT_THAT(resp, ArgType(RespExpr::ARRAY));
+  EXPECT_THAT(resp.GetVec(), ElementsAre(IntArg(0), IntArg(1), IntArg(2), IntArg(3)));
+
+  json = R"(
+    [[], "a", ["a", "b"], ["a", "b", "c"], 4]
+  )";
+
+  resp = Run({"set", "json", json});
+  ASSERT_THAT(resp, "OK");
+
+  resp = Run({"JSON.ARRLEN", "json", "$[*]"});
+  ASSERT_THAT(resp, ArgType(RespExpr::ARRAY));
+  EXPECT_THAT(resp.GetVec(), ElementsAre(IntArg(0), ArgType(RespExpr::NIL), IntArg(2), IntArg(3), ArgType(RespExpr::NIL)));
+}
+
+TEST_F(JsonFamilyTest, Toggle) {
+  string json = R"(
+    {"a":true, "b":false, "c":1, "d":null, "e":"foo", "f":[], "g":{}}
+  )";
+
+  auto resp = Run({"set", "json", json});
+  ASSERT_THAT(resp, "OK");
+
+  resp = Run({"JSON.TOGGLE", "json", "$.*"});
+  ASSERT_THAT(resp, ArgType(RespExpr::ARRAY));
+  EXPECT_THAT(resp.GetVec(), ElementsAre(IntArg(0), IntArg(1), ArgType(RespExpr::NIL), ArgType(RespExpr::NIL), ArgType(RespExpr::NIL), ArgType(RespExpr::NIL), ArgType(RespExpr::NIL)));
+
+  resp = Run({"JSON.GET", "json", "$.*"});
+  EXPECT_EQ(resp, string_view(R"([false,true,1,null,"foo",[],{}])", 31));
```

Comment:
why did you need to wrap it with explicit string_view c'tor?

Reply:
Thanks for this comment. It's definitely redundant, so I removed those explicit string_view c'tors.